docs/rules/no-this-property-during-ssr.md

```diff
@@ -0,0 +1,48 @@
+# Disallow access of properties on this during SSR (`lwc/no-this-property-during-ssr`)
```

Comment:
I would like to better understand the motivation behind this rule. It appears to me that we are bundling multiple potential SSR errors in a single rule:
- Access to certain DOM APIs that aren't implemented by the LWC engine server: eg. `dispatchEvent`
- Access to rendered content prior to the component being rendered `this[.template].querySelector`
- Access to the parent node via the template `this.template.host`
- Access to DOM APIs poly filled on the LWC engine server: eg. `this.getAttribute`, `this.setAttribute`, `this.classList`

Could we tease those use cases apart and evaluate the usefulness of each of them individually?